src/context.cc

```diff
@@ -1373,48 +1373,45 @@ error Context::downloadUpdate() {
             return err;
         }
 
-        // Get update check URL
-        std::string update_url("");
-        err = updateURL(&update_url);
-        if (err != noError) {
-            return err;
+        if (HTTPSClient::Config.AppVersion.empty()) {
```

Comment:
Here I'd probably reword the error. This is good for us as the developers but if there's a broken version somewhere in the wild, the user won't know what's going on. I'd go for something like "This version cannot check for updates. This has been probably already fixed. Please check toggl.com for a newer version."
Also, reporting to bugsnag in this place would be a good idea so we know immediately.

Reply:
I think these are errors for developers, as they shouldn't happen for users (definitely not in beta or stable) so we're optimizing the developer-fix time instead of trying to explain stuff to the users. We have plenty of such errors elsewhere; actually, I just moved this piece of code. And it does get reported to Bugsnag, at least for Windows we subscribe to `toggl_on_error`.

Reply:
If the message gets displayed in the UI (and this message will), the user will see it and get confused by it. 
It's not for optimization of the fix time (and making it more comprehensible doesn't change anything for us because we can still `grep` the string in the source code)

Reply:
Of course, the chance of this happening is **slim** but still, we can start handling app versions different in the future, or this can happen to somebody who got the app from some other source than us (like unofficial packages on Linux).

Reply:
Updated the error message, @MartinBriza